node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeagent/NodeAgentImpl.java

```diff
@@ -469,6 +468,12 @@ void converge() {
         if (!node.equals(lastNode)) {
             logChangesToNodeSpec(lastNode, node);
 
+            if (currentRebootGeneration < node.getCurrentRebootGeneration())
```

Comment:
These two ifs are for initializing the generations? Seems to me that other than the first time, we should never set the current generations from the node repo.

Reply:
That's right.